docs/references/java_project/index.md

```diff
@@ -1,17 +1,96 @@
+# Overview 
+
+
+### Templates
+
+All the templates generate the base project structure along with some additional files and directories specific to 
+Java (highlighted below): 
+
+``` hl_lines="3 4 35-38"
+.
+├── Dockerfile
+├── build.gradle.kts
+├── commands.cfg
+├── connections.json
+├── eula.txt
+├── manifest.txt
+├── requirements.txt
+├── conf
+│   └── describeSchema.xsd
+├── content
+│   ├── alertdefs
+│   │   └── alertDefinitionSchema.xsd
+│   ├── customgroups
+│   ├── dashboards
+│   ├── files
+│   │   ├── reskndmetric
+│   │   ├── solutionconfig
+│   │   ├── topowidget
+│   │   └── txtwidget
+│   ├── policies
+│   ├── recommendations
+│   ├── reports
+│   ├── resources
+│   ├── supermetrics
+│   ├── symptomdefs
+│   └── traversalspecs
+│       └── traversalSpecsSchema.xsd
+├── images
+│   ├── AdapterKind
+│   ├── ResourceKind
+│   └── TraversalSpec
+├── resources
+│   └── resources.properties
+└── src
+    └── com
+        └── mycompany
+            └── Adapter.java
+```
+
+### src (directory)
+
+This directory contains the package structure specified during `mp-init`. Inside the package, every template creates an
+Adapter.java file that serves as the main class for the project. 
+
+### build.gradle.kts (file)
+
+This files contains all the project dependencies along with some building specifications used by the Dockerfile. 
+
+### commands.cfg (file)
+This file contains a list of the commands the HTTP server can run, along with the path to the executable related to the
+command. By default, all commands are run by executing the `adapter.py` file along with a parameter that defines a command.
+For example, when the HTTP server receives a request to run a test connection, it reads the commands.cfg key for `test`
+and runs the process defined by the key value, `/usr/local/bin/python app/adapter.py test`.
```

Comment:
We should adapt this to the specifics of Java, or just put it in the base documentation